### ISSUE TITLE
max_long_positions=2

### ISSUE THREAD

**Issue by markxbaker** [edtechre/pybroker#182]
Hey @edtechre 

i have this set but it opens more positions than expected, am i using wrong or is a bug? sorry for bomabrdment!
I only have one instrument as part of  the test but seems to open several overlapping trades

# Backtest configuration
config = StrategyConfig(
    initial_cash=998_270,
    fee_mode=FeeMode.PER_SHARE,
    fee_amount=0.02,
    max_long_positions=2,
    return_signals=True,
    return_stops=True,
    round_fill_price=False

)

# Execution function
def exec_fn(ctx: ExecContext):
    
    atr_10d = ctx.indicator('ATR_10')

    ATR = 4.5 * atr_10d[-1]
    long_pos = ctx.long_pos()
   
    cross_up = ctx.close[-1] > ctx.open[-1]
 
    # Check for entry condition
    if cross_up:

          ctx.buy_shares = 1
          ctx.buy_fill_price = PriceType.OPEN
          ctx.sell_fill_price = PriceType.OPEN
          ctx.hold_bars = 5
          ctx.stop_loss = ATR

# Set up strategy and run backtest
strategy = Strategy(data_source_M1, start_date, end_date, config)
strategy.add_execution(exec_fn, tickers, indicators=list(indicators.values()))

# Run backtest
result = strategy.backtest(warmup=10)

pd.options.display.float_format = '{:,.5f}'.format
long_trades = result.trades
long_trades

![Image](https://github.com/user-attachments/assets/035cd778-ca14-4f6c-808b-298480d52849)
